numpy/distutils/conv_template.py

```diff
@@ -317,21 +317,24 @@ def main():
     try:
         file = sys.argv[1]
     except IndexError:
-        fid = sys.stdin
-        outfile = sys.stdout
+        fid = contextlib_nullcontext(sys.stdin)
+        outfile = contextlib_nullcontext(sys.stdout)
     else:
         fid = open(file, 'r')
         (base, ext) = os.path.splitext(file)
         newname = base
         outfile = open(newname, 'w')
 
-    allstr = fid.read()
+    with fid:
+        allstr = fid.read()
     try:
         writestr = process_str(allstr)
     except ValueError:
         e = get_exception()
         raise ValueError("In %s loop at %s" % (file, e))
-    outfile.write(writestr)
+
+    with outfile:
```

Comment:
Hmm, unfortunately the handling of this function still isn't correct, and without `contextlib.ExitStack` it's really hard to get right. As an example, if `open(newname)` fails then `open(file)` is not closed.

Can you revert the changes to the two functions in this patch that do this `stdin` / `stdout` juggling, and leave them to a later PR? I think everything else in this patch is trivially correct, so would be good to get in first.

Reply:
Done